client/sender.go

```diff
@@ -85,3 +86,46 @@ func newSender(u *url.URL, ctx *base.Context, retryOptions retry.Options, stoppe
 	}
 	return f(u, ctx, retryOptions, stopper)
 }
+
+// SendWrapped is a convenience function which wraps the request in a batch,
+// sends it via the provided Sender, and returns the unwrapped response
+// or an error. It's valid to pass a `nil` context; context.TODO() is used
+// in that case.
+func SendWrapped(sender Sender, ctx context.Context, args roachpb.Request) (roachpb.Response, error) {
+	if ctx == nil {
+		ctx = context.TODO()
+	}
+	ba, unwrap := func(args roachpb.Request) (*roachpb.BatchRequest, func(*roachpb.BatchResponse) roachpb.Response) {
+		ba := &roachpb.BatchRequest{}
+		{
+			h := *(proto.Clone(args.Header()).(*roachpb.RequestHeader))
+			ba.Key, ba.EndKey = h.Key, h.EndKey
+			ba.CmdID = h.CmdID
+			ba.Timestamp = h.Timestamp
+			ba.Replica = h.Replica
+			ba.RangeID = h.RangeID
+			ba.UserPriority = h.UserPriority
+			ba.Txn = h.Txn
+			ba.ReadConsistency = h.ReadConsistency
+		}
+		ba.Add(args)
+		return ba, func(br *roachpb.BatchResponse) roachpb.Response {
+			unwrappedReply := br.Responses[0].GetInner()
+			// The ReplyTxn is propagated from one response to the next request,
+			// and we adopt the mechanism that whenever the Txn changes, it needs
+			// to be set in the reply, for example to ratchet up the transaction
+			// timestamp on writes when necessary.
+			// This is internally necessary to sequentially execute the batch,
+			// so it makes some sense to take the burden of updating the Txn
+			// from TxnCoordSender - it will only need to act on retries/aborts
+			// in the future.
+			unwrappedReply.Header().Txn = br.Txn
+			return unwrappedReply
+		}
+	}(args)
+	br, pErr := sender.Send(context.TODO(), *ba)
```

Comment:
comedy equals sloppyness plus linters that don't flag variables as unused when they're function parameters. Fixed.